src/test/java/uk/gov/laa/gpfd/dao/ReportViewsDaoTest.java

```diff
@@ -0,0 +1,202 @@
+    }
+
+    @Test
+    void shouldHandleInvalidQueryWithSyntaxError() {
+        String sqlQuery = "INVALID SQL QUERY";
+        when(writeJdbcTemplate.queryForList(sqlQuery)).thenThrow(new DataAccessException("Syntax error") {});
+
+        DatabaseReadException exception = assertThrows(DatabaseReadException.class,
+                () -> reportViewsDao.callDataBase(sqlQuery));
+        assertEquals("Error reading from DB: uk.gov.laa.gpfd.dao.ReportViewsDaoTest$2: Syntax error", exception.getMessage());
+    }
+
+    @Test
+    void shouldReturnMultipleRowsSuccessfully() {
+        String sqlQuery = "SELECT * FROM reports";
+        List<Map<String, Object>> mockResult = List.of(Map.of("column1", "value1"), Map.of("column1", "value2"));
+        when(writeJdbcTemplate.queryForList(sqlQuery)).thenReturn(mockResult);
+
+        List<Map<String, Object>> result = reportViewsDao.callDataBase(sqlQuery);
+
+        assertNotNull(result);
+        assertEquals(2, result.size());
+        assertEquals("value1", result.get(0).get("column1"));
+        assertEquals("value2", result.get(1).get("column1"));
+    }
+
+    @Test
+    void shouldHandleSpecialCharactersInResult() {
+        String sqlQuery = "SELECT * FROM reports";
+        List<Map<String, Object>> mockResult = List.of(Map.of("column1", "!@#$%^&*()"));
+        when(writeJdbcTemplate.queryForList(sqlQuery)).thenReturn(mockResult);
+
+        List<Map<String, Object>> result = reportViewsDao.callDataBase(sqlQuery);
+
+        assertNotNull(result);
+        assertEquals("!@#$%^&*()", result.get(0).get("column1"));
+    }
+
+    @Test
+    void shouldThrowExceptionForInvalidDataType() {
+        String sqlQuery = "SELECT * FROM reports";
+        when(writeJdbcTemplate.queryForList(sqlQuery)).thenThrow(new ClassCastException("Cannot cast"));
+
+        ClassCastException exception = assertThrows(ClassCastException.class,
+                () -> reportViewsDao.callDataBase(sqlQuery));
+        assertEquals("Cannot cast", exception.getMessage());
+    }
+
+    @Test
+    void shouldThrowExceptionForEmptyQuery() {
+        String sqlQuery = "";
+        DatabaseReadException exception = assertThrows(DatabaseReadException.class,
+                () -> reportViewsDao.callDataBase(sqlQuery));
+        assertEquals("No results returned from query to MOJFIN reports database", exception.getMessage());
+    }
+
+    @Test
+    void shouldRetrieveDataWithValidHeaders() {
+        String sqlQuery = "SELECT * FROM reports";
+        List<Map<String, Object>> mockResult = List.of(Map.of("column1", "value1"));
+        when(writeJdbcTemplate.queryForList(sqlQuery)).thenReturn(mockResult);
+
+        List<Map<String, Object>> result = reportViewsDao.callDataBase(sqlQuery);
+
+        assertNotNull(result);
+        assertEquals("value1", result.get(0).get("column1"));
+    }
+
+    @Test
+    void shouldRetrieveDataWithSpecificParameters() {
```

Comment:
This isn't testing that data is retrieved using specific parameters, its testing that you can send sql with parameters. 

shouldAllowQueriesWithParameters perhaps